lib/config/src/package/mod.rs

```diff
@@ -0,0 +1,1490 @@
+    )]
+    pub disable_command_rename: bool,
+    /// Unlike, `disable-command-rename` which prevents `wasmer run <Module name>`,
+    /// this flag enables the command rename of `wasmer run <COMMAND_NAME>` into
+    /// just `<COMMAND_NAME>`. This is useful for programs that need to inspect
+    /// their `argv[0]` names and when the command name matches their executable
+    /// name.
+    #[serde(
+        rename = "rename-commands-to-raw-command-name",
+        default,
+        skip_serializing_if = "std::ops::Not::not"
+    )]
+    #[builder(default)]
+    #[deprecated(
+        since = "0.9.2",
+        note = "Does nothing. Prefer a runner-specific command attribute instead"
+    )]
+    pub rename_commands_to_raw_command_name: bool,
+    /// The name of the command that should be used by `wasmer run` by default.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[builder(setter(into, strip_option), default)]
+    pub entrypoint: Option<String>,
+    /// Mark this as a private package
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    #[builder(default)]
+    pub private: bool,
+}
+
+impl Package {
+    /// Create a [`PackageBuilder`] populated with all mandatory fields.
+    pub fn builder(
+        name: impl Into<String>,
+        version: Version,
+        description: impl Into<String>,
+    ) -> PackageBuilder {
+        PackageBuilder::new(name, version, description)
+    }
+}
+
+impl PackageBuilder {
+    pub fn new(name: impl Into<String>, version: Version, description: impl Into<String>) -> Self {
+        let mut builder = PackageBuilder::default();
+        builder.name(name).version(version).description(description);
+        builder
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum Command {
+    V1(CommandV1),
+    V2(CommandV2),
+}
+
+impl Command {
+    /// Get the command's name.
+    pub fn get_name(&self) -> &str {
+        match self {
+            Self::V1(c) => &c.name,
+            Self::V2(c) => &c.name,
+        }
+    }
+
+    /// Get the module this [`Command`] refers to.
+    pub fn get_module(&self) -> &ModuleReference {
+        match self {
+            Self::V1(c) => &c.module,
+            Self::V2(c) => &c.module,
+        }
+    }
+}
+
+/// Describes a command for a wasmer module.
+///
+/// When a command is deserialized using [`CommandV1`], the runner is inferred
+/// by looking at the [`Abi`] from the [`Module`] it refers to.
+///
+/// If possible, prefer to use the [`CommandV2`] format.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)] // Note: needed to prevent accidentally parsing
+// a CommandV2 as a CommandV1
+#[deprecated(since = "0.9.2", note = "Prefer the CommandV2 syntax")]
+pub struct CommandV1 {
+    pub name: String,
+    pub module: ModuleReference,
+    pub main_args: Option<String>,
+    pub package: Option<String>,
+}
```

Comment:
Should we just make commandv2 the base and remove v1?

Reply:
I'd rather not do any cleanup right now, let's leave that as a followup?

Reply:
https://github.com/wasmerio/wasmer/issues/4584